Sources/PackageModel/UserToolchain.swift

```diff
@@ -142,7 +143,12 @@ public final class UserToolchain: Toolchain {
         {
           return librarian.basename
         }
-        // TODO(5719) use `lld-link` if the build requests lld.
+        // TODO(5719) handle `-Xmanifest` vs `-Xswiftc`
+        // `-use-ld=` is always joined in Swift.
+        if let ld = extraSwiftFlags.first(where: { $0.starts(with: "-use-ld=") }) {
```

Comment:
string parsing seems fragile, is there a better way?

Reply:
I really don't see a better way unfortunately.  The driver flags, frontend flags, and non-Swift tool flags are all intertwined and will impact this.  I think that this might be the best approach (at least for now).

Reply:
https://github.com/apple/swift-package-manager/pull/5981/commits/d77f51d6c15d4ea63cece9793008d6838d93ee56#diff-adf7999cea9f76ad6a7d320c04fbfa7222d4c7d20bd27f81667c02cf88081897R140 BTW is where this becomes Windows specific.